thrift_span.go

```diff
@@ -86,24 +87,28 @@ func buildAnnotations(span *span, endpoint *z.Endpoint) []*z.Annotation {
 	}
 	if !span.startTime.IsZero() && startLabel != "" {
 		start := &z.Annotation{
-			Timestamp: timeToMicrosecondsSinceEpochInt64(span.startTime),
+			Timestamp: utils.TimeToMicrosecondsSinceEpochInt64(span.startTime),
 			Value:     startLabel,
 			Host:      endpoint}
 		annotations = append(annotations, start)
 		if span.duration != 0 {
 			endTs := span.startTime.Add(span.duration)
 			end := &z.Annotation{
-				Timestamp: timeToMicrosecondsSinceEpochInt64(endTs),
+				Timestamp: utils.TimeToMicrosecondsSinceEpochInt64(endTs),
 				Value:     endLabel,
 				Host:      endpoint}
 			annotations = append(annotations, end)
 		}
 	}
 	for _, log := range span.logs {
 		anno := &z.Annotation{
-			Timestamp: timeToMicrosecondsSinceEpochInt64(log.Timestamp),
-			Value:     truncateString(log.Event),
+			Timestamp: utils.TimeToMicrosecondsSinceEpochInt64(log.Timestamp),
 			Host:      endpoint}
+		if content, err := spanlog.MaterializeWithJSON(log.Fields); err == nil {
+			anno.Value = truncateString(string(content))
```

Comment:
valid point, but we do this with other data as well, e.g. if the user sends a JSON as a tag. Ideally we should have some sort of mechanism in the collection pipeline that does intelligent throttling of the tags/logs.